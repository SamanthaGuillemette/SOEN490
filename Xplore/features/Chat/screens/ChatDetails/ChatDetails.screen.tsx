import { NavigationProp } from "@react-navigation/native";
import { useRoute } from "@react-navigation/native";
import { SafeAreaView } from "react-native";
import { useThemeColor } from "../../../../hooks/useThemeColor";
import ChatDetailsHeader from "../../components/ChatDetailsHeader/ChatDetailsHeader.Comonent";
import Conversation from "../../components/Conversation/Conversation.Component";
import ChatTextInput from "../../components/ChatTextInput/ChatTextInput.Component";
import styles from "./ChatDetails.styles";
interface ChatDetailsProps {
  navigation: NavigationProp<any>;
}

const ChatDetails = (props: ChatDetailsProps) => {
  const route = useRoute();
  let { name }: any = route.params;
  const backgroundSecondary = useThemeColor("backgroundSecondary");
  return (
    <SafeAreaView
      style={[styles.safeAreaStyle, { backgroundColor: backgroundSecondary }]}
    >
      <ChatDetailsHeader username={name} navigation={props.navigation} />
      <Conversation navigation={name} />
      <ChatTextInput />
    </SafeAreaView>
  );
};

export default ChatDetails;
