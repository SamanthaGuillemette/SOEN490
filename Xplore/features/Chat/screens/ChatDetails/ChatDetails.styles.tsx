import { StyleSheet, StatusBar } from "react-native";

export default StyleSheet.create({
  safeAreaStyle: {
    flex: 1,
    marginTop: StatusBar.currentHeight ?? 0,
    marginBottom: 15,
  },
  messages_container: {
    flex: 1,
    marginTop: 10,
  },
});
