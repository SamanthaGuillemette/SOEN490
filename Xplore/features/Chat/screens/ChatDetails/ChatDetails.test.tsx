import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import ChatDetails from "./ChatDetails.screen";

// Create stack navigator
const Stack = createNativeStackNavigator();

// Wrap test component in navigation container
function ChatDetailsWithNavigation() {
  return (
    <NavigationContainer>
      <Stack.Screen name="ChatDetails" component={ChatDetails} />
    </NavigationContainer>
  );
}

describe("Tesing for <ChatDetails /> screen:", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<ChatDetailsWithNavigation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly", () => {
    const container = render(<ChatDetailsWithNavigation />);
    expect(container).toBeTruthy();
  });
});
