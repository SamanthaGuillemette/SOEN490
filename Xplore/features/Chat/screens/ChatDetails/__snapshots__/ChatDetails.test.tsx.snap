// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tesing for <ChatDetails /> screen: matches snapshot 1`] = `null`;
